### PR TITLE
Added mDNS client/server

### PR DESCRIPTION
**Pull request by TurkeyMan**

### CODE REVIEW

File: src/protocol/dns/message.d [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    {
+        MutableString!0 qname;
+        if (!parseName(qname, msg, data) && msg.length >= 4)
+            return false;
+
+        DNSType qtype = msg.takeFront!2.bigEndianToNative!DNSType;
+        DNSClass qclass = msg.takeFront!2.bigEndianToNative!DNSClass;
+        ushort highBit = qclass & 0x8000;
+        qclass ^= highBit;
+        bool preferUnicastResponse = highBit != 0;
```
Comment (megaeels) [inline src/protocol/dns/message.d:143]:
```suggestion
        bool preferUnicastResponse = (qclass & 0x8000) != 0;
        qclass &= 0x7FFF;
```

File: src/protocol/dns/message.d [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    {
+        MutableString!0 name;
+        if (!parseName(name, msg, data) && msg.length >= 10)
+            return false;
+
+        DNSType type = msg.takeFront!2.bigEndianToNative!DNSType;
+        DNSClass cls = msg.takeFront!2.bigEndianToNative!DNSClass;
+        ushort highBit = cls & 0x8000;
+        cls ^= highBit;
+        bool flushCache = highBit != 0;
```
Comment (megaeels) [inline src/protocol/dns/message.d:158]:
```suggestion
        bool flushCache = (cls & 0x8000) != 0;
        cls &= 0x7FF;
```

File: src/protocol/dns/package.d [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+//            foreach(i; 0 .. _interface.length)
+//            {
+//                BaseInterface iface = app.moduleInstance!InterfaceModule.findInterface(_interface[i]);
+//                if (!iface)
+//                {
+//                    session.writeLine("Interface does not exist: ", _interface[i]);
+//                    return;
+//                }
+//                interfaces ~= iface;
+//            }
```
Comment (megaeels) [inline src/protocol/dns/package.d:54]: wip comments

File: src/protocol/dns/package.d [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+            NoGCAllocator a = app.allocator;
+
+            String n = name.makeString(a);
+            mDNSServer server = a.allocT!mDNSServer(n.move);//, interfaces.move);
```
Comment (megaeels) [inline src/protocol/dns/package.d:59]: Here too.

File: src/urt/inet.d [hunk 5]
```diff
@@ -16,6 +16,7 @@ enum AddressFamily : byte
 	IPv6,
 }
 
+enum IPAddr IPAddrLit(string addr) = () { IPAddr a; size_t taken = a.fromString(addr); assert(taken == addr.length, "Not an IPV4 address"); return a; }();
```
Comment (megaeels) [inline src/urt/inet.d:19]: There's one too many things going on here.  Explain.

**Review by megaeels — COMMENTED** [review]
It looks like this work isn't quite complete, lots of comments that should be removed.

File: src/urt/inet.d (hunk 5 above)
Comment (TurkeyMan) [reply src/urt/inet.d:19]: The `enum` is assigned the result of the CTFE function that parses the IP address from the given string (using the normal runtime function); there is a lambda, which is immediately called (final parens) to produce a result.

**Comment by TurkeyMan** [comment]
Yeah, I shouldn't have pushed this I guess... I was trying to get it out of my merge queue, and it was next in line with no dependencies.

**Pull request merged**